DownFall/Views/Button.swift

```diff
@@ -128,40 +165,80 @@ class Button: SKSpriteNode {
 
 extension Button {
     override func touchesEnded(_ touches: Set<UITouch>, with event: UIEvent?) {
+        guard let touch = touches.first else { return }
+        let position = touch.location(in: self)
+        let translatedPosition = CGPoint(x: self.frame.center.x + position.x, y: self.frame.center.y + position.y)
+        if self.frame.contains(translatedPosition) {
+            buttonWasTapped()
+        } else {
+            buttonTapWasCancelled()
+        }
+    }
```

Comment:
The frame should be adjusted to account for the button movement.  If a player starts/ends a tap in the top `dropShadowOffet` points of the button, this gets recorded as a cancel.  In fact, it would be nice to use this method to create an larger touch zone than the button itself, if space permits.